File: GAMEOFLIFE/DeadlyNightshade.cpp
#include "DeadlyNightshade.h"
#include "World.h"

DeadlyNightshade::DeadlyNightshade(int x, int y) : Plant('X', 10, x, y, 99, 0) {
}

void DeadlyNightshade::Collision(Organism* other) {
	world->AddLog(this, "poisoned an animal and died", 2);
	other->Kill();
	Kill();
}

Organism* DeadlyNightshade::Child() {
	return new DeadlyNightshade(*this);
}

File: GAMEOFLIFEJAVA/gameoflife/organisms/plants/DeadlyNightshade.java
package gameoflife.organisms.plants;

import gameoflife.structs.Vector;
import gameoflife.organisms.Organism;

public class DeadlyNightshade extends Plant {

    public DeadlyNightshade(Vector position){
        super("%", 99, position);
    }

    @Override
    public void Collision(Organism other){

        world.Log(this.getClass().getSimpleName() +" poisoned a " + other.getClass().getSimpleName() + " and died...");
        other.Kill();
        this.Kill();
    }
    
}
